lib/matplotlib/tests/test_legend.py

```diff
@@ -1072,7 +1073,7 @@ def test_legend_labelcolor_rcparam_markerfacecolor_short():
 
 
 def test_get_set_draggable():
-    legend = plt.legend()
+    legend = plt.legend(labels=["mock data"])
```

Comment:
I'm somewhat uneasy about this pattern of providing labels when there is no data.

1. Passing labels only is supported but discouraged.
2. While currently supported, it feels, that passing more labels than we have data should possibly warn as well. At least it's non-sensical.

I suggest to silence the warning explicitly rather than through a sketchy code construct: Decorate the function with
```
@pytest.mark.filterwarnings("ignore:No artists with labels found to put in legend")
```
instead.

Also in the following test functions.